adapters.go

```diff
@@ -33,13 +35,18 @@ type whisperAdapter struct {
 	privateKey *ecdsa.PrivateKey
 	transport  transport.WhisperTransport
 	protocol   *encryption.Protocol
+	logger     *zap.Logger
```

Comment:
I am not fan of putting it inside a struct because it is not part of the business logic. Maybe we can just create a global unexported variable? Like
```go
var logger = logger.With(zap.Namespace("whisperAdapter"))
```

Initial `logger` instance can be taken from `log.go`.

Reply:
I thought we'd avoid using global variables (it's also noted in the zap documentation that while supported, they only [advise](https://github.com/uber-go/zap/blob/master/FAQ.md#why-include-package-global-loggers) using the logger as a global variable for legacy apps which can't afford to change APIs). But if that's the way we want to do it, I can move to a global logger.

Reply:
In a sense, a Mutex is not part of the business logic either, it's an implementation detail.

Reply:
> Avoid them where possible.

It would be great to see some explanation :D I understand that it might be beneficial for testing or when the logger is different for instances.

I can live with this and let's keep it, just don't see any value now.

Reply:
> In a sense, a Mutex is not part of the business logic either, it's an implementation detail.

That's a fair point, however, usually, Mutex is not a part of the business logic. However, in this case, probably the even bigger thing is the fact that `logger` is in the constructor params. If that's unexported constructor, fine. But we need to be careful about exported ones.

Reply:
> It would be great to see some explanation :D I understand that it might be beneficial for testing or when the logger is different for instances.

Yeah, easier testing is one of the reasons. Also we'd like to have a different instance for each class because each instance will contain additional context (like the class name it is logging for, and potentially other context in the future. E.g. for testing we're adding the user name, Alice, Bob, etc)